highlight/highlight
# Fix bugs applying filters from log/span attributes

## Summary

Fixes a few issues around applying attribute filters.

1. Applying attribute filters that contained `"` was broken because a) we didn't wrap the value in quotes properly, and b) our query language doesn't support escaping quotes.
2. Removes the `message` prefix when applying an attribute filter on logs since this would break the query.
3. Nothing would happen when trying to apply a `message` attribute filter when there was no query present.

## How did you test this change?

1. Apply a `message` filter and confirm the `message=` prefix is not applied
1. Apply a `message` filter when the query input is empty
1. Apply a filter that contains quoted strings within it

Confirmed all broken behavior was working as expected locally, in prod, and on the customer's project who reported the error.

## Are there any deployment considerations?

Should follow up with customer who reported the issue.

## Does this work require review from our design team?

N/A
